src/engine/util/queue.ts

```
@@ -47,7 +47,13 @@ export class Queue<TItem> {
      * @returns The item removed.
      */
     public dequeue(): TItem {
-        return this._items.shift();
+        const item = this._items.shift();
+
+        if (!item) {
+            throw new Error('Cannot dequeue an empty queue.');
```

Do we want to throw here of just return `TItem | undefined` like for example `Array.at` does?
In case we do want to throw, it should probably be documented that the method *CAN* throw.